Load error with duplicate .ji files

Hey. This morning, I got the following error when opening a julia prompt:

~~~
$ julia
               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.3.1 (2019-12-30)
 _/ |\__'_|_|_|\__'_|  |  Official https://julialang.org/ release
|__/                   |

┌ Warning: Could not load Revise
│   Error =
│    InitError: AssertionError: length(paths) == 1
│    during initialization of module Revise
└ @ Main ~/.julia/config/startup.jl:6
(v1.3) pkg> st
    Status `~/.julia/environments/v1.3/Project.toml`
  [295af30f] Revise v2.5.1 [`~/.julia/dev/Revise`]
~~~

`paths` was `["/home/felix/.julia/compiled/v1.3/JuliaInterpreter/PliIn_Tzgkg.ji", "/home/felix/.julia/compiled/v1.3/JuliaInterpreter/PliIn_dWBvD.ji"]` immediately before, so I moved the `compiled/v1.3/JuliaInterpreter` folder and it went away.

Not sure how I got into this situation, but I guess there is a bug somewhere?

--- a/src/pkgs.jl
+++ b/src/pkgs.jl
@@ -96,18 +96,13 @@ function pkg_fileinfo(id::PkgId)
     sourcepath = Base.locate_package(id)
     if length(paths) > 1
         fpaths = filter_valid_cachefiles(sourcepath, paths)
-        if isempty(fpaths)
-            # Work-around for #371 (broken dependency prevents tracking):
-            # find the most recent cache file. Presumably this is the one built
-            # to load the package.
-            sort!(paths; by=path->mtime(path), rev=true)
-            deleteat!(paths, 2:length(paths))
-        else
-            paths = fpaths
-        end
+        paths = isempty(fpaths) ? paths : fpaths
+        # Work-around for #371 (broken dependency prevents tracking):
+        # find the most recent cache file. Presumably this is the one built
+        # to load the package.
+        sort!(paths; by=path->mtime(path), rev=true)
     end
     isempty(paths) && return nothing, nothing
-    @assert length(paths) == 1
     path = first(paths)
     provides, includes_requires = try
         parse_cache_header(path)

